File: src/TypeFormatter/AliasTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import { AliasType } from "../Type/AliasType.js";
import type { BaseType } from "../Type/BaseType.js";
import type { TypeFormatter } from "../TypeFormatter.js";

export class AliasTypeFormatter implements SubTypeFormatter {
    public constructor(protected childTypeFormatter: TypeFormatter) {}

    public supportsType(type: BaseType): boolean {
        return type instanceof AliasType;
    }
    public getDefinition(type: AliasType): Definition {
        return this.childTypeFormatter.getDefinition(type.getType());
    }
    public getChildren(type: AliasType): BaseType[] {
        return this.childTypeFormatter.getChildren(type.getType());
    }
}

File: internal/formatter/alias_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// AliasTypeFormatter mirrors src/TypeFormatter/AliasTypeFormatter.ts.
type AliasTypeFormatter struct {
	childTypeFormatter TypeFormatter
}

func NewAliasTypeFormatter(childTypeFormatter TypeFormatter) *AliasTypeFormatter {
	return &AliasTypeFormatter{childTypeFormatter: childTypeFormatter}
}

func (f *AliasTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.AliasType)
	return ok
}

func (f *AliasTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return f.childTypeFormatter.GetDefinition(t.(*types.AliasType).Type)
}

func (f *AliasTypeFormatter) GetChildren(t types.Type) []types.Type {
	return f.childTypeFormatter.GetChildren(t.(*types.AliasType).Type)
}
